pandas/core/internals/managers.py

```diff
@@ -1952,8 +1952,25 @@ def _compare_or_regex_search(a, b, regex=False):
         # GH#29553 avoid deprecation warnings from numpy
         result = False
```

Comment:
let's just return here (you will need move the check on like 194 into a function ,e.g.

```
if is_datetimeliek......:
     return check(False)

.....
```

Reply:
Hi @jreback, I'm not sure if I completely understood the change you requested to make here. But I assume you wanted to place the entire block that checks the result's value into a function instead? If so, I did that and made some additional changes in order to have it working. Can you please confirm? Thanks